content/doc/book/pipeline/jenkinsfile.adoc

```diff
@@ -403,18 +411,383 @@ apply to all steps within the Pipeline.
+====
+
+The following code snippet shows the example Pipeline in its entirety:
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent {
+        // Define agent details here
+    }
+    stages {
+        stage('Example stage 1') {
+            environment {
+                BITBUCKET_COMMON_CREDS = credentials('jenkins-bitbucket-common-creds')
+            }
+            steps {
+                // // <1>
+            }
+        }
+        stage('Example stage 2') {
+            steps {
+                // // <2>
+            }
+        }
+    }
+}
+// Script //
+----
+<1> The following credential environment variables (defined in this Pipeline's
+<<syntax#environment,`environment`>> directive) are available within this
+stage's steps and can be referenced using the syntax:
+* `$BITBUCKET_COMMON_CREDS`
+* `$BITBUCKET_COMMON_CREDS_USR`
+* `$BITBUCKET_COMMON_CREDS_PSW`
+
++
+For example, here you can authenticate to Bitbucket with the username and
+password assigned to these credential variables. +
+To maintain the security and anonymity of these credentials, if you attempt to
+retrieve the value of these credential variables from within the Pipeline, the
+same behavior described in the <<#secret-text,Secret text>> example above
+applies to these username and password credential variable types too.
+<2> In this Pipeline example, the credentials assigned to the three
+`COMMON_BITBUCKET_CREDS...` environment variables are scoped only to `Example
+stage 1`, so these credential variables are not available for use in this
+`Example stage 2` stage's steps. If, however, the `environment` directive in
+this Pipeline were moved immediately within the <<syntax#declarative-pipeline,
+`pipeline`>> block (as is the case in the <<#secret-text,Secret text>> Pipeline
+example above), then these `COMMON_BITBUCKET_CREDS...` environment variables
+would be scoped globally and could be used in any stage's steps.
+
+
+===== Secret files
+
+As far as Pipelines are concerned, secret files are handled in exactly the same
+manner as secret text (<<#secret-text,above>>).
+
+Essentially, the only difference between secret text and secret file credentials
+are that for secret text, the credential itself is entered directly into Jenkins
+whereas for a secret file, the credential is originally stored in a file which
+is then uploaded to Jenkins.
+
+Unlike secret text, secret files cater for credentials that are:
+
+* too unwieldy to enter directly into Jenkins, and/or
+* in binary format, such as a GPG file.
+
+
+==== For other credential types
+
+If you need to set credentials in a Pipeline for anything other than secret
+text, usernames and passwords, or secret files
+(<<#for-secret-text-usernames-and-passwords-and-secret-files,above>>) - i.e SSH
+keys or certificates, then use Jenkins' *Snippet Generator* feature, which you
+can access through Jenkins' classic UI.
+
+To access the *Snippet Generator* for your Pipeline project/item:
```

Comment:
Do we need a general Snippet Generator page/section?  If you had that, this section could be a little less detailed.   Probably need a separate PR/task to cover that change. 

As it stands, the is good, but would need to change if/when the Snippet Generate pages is created.

Reply:
Yes that's a good point ... when I next include a procedure that makes use of the Snippet Generator on this 'Using a Jenkinsfile' page (or another other relevant page in the 'Pipeline' chapter), then I'll extract this information into its own section near the top of this 'Using a Jenkinsfile' page.